viewing_party/party.py

```diff
@@ -0,0 +1,173 @@
+        return None
+
+    for item in user_data["watched"]:
+        if (item["genre"] in genre_count):
+            genre_count[item["genre"]] += 1
+        else:
+            genre_count[item["genre"]] = 1
+    
+    most_frequent_genre = max(genre_count, key=genre_count.get)
+
+    return most_frequent_genre
+
+def get_unique_watched(user_data):
+    unique_watched = []
+    friends = user_data["friends"]
+    friends_watched_list = []
+    i = 0
+    
+    for i in range(len(friends)):
+        friends_watched = friends[i]["watched"]
+        for i in range(len(friends_watched)):
+            if friends_watched[i] not in friends_watched_list:
+                friends_watched_list.append(friends_watched[i])
+    
+    for item in user_data["watched"]:
+        if item not in friends_watched_list:
+            unique_watched.append({"title" : item["title"]})
+
+    return unique_watched
+
+    
+def get_friends_unique_watched(user_data):
+    unique_unwatched = []
+    user = user_data["watched"]
+    friends = user_data["friends"]
+    user_watched = []
+    i = 0
+    
+    for i in range(len(user)):
+        if user[i] not in user_watched:
+            user_watched.append(user[i])
+    
+    for i in range(len(friends)):
+        friends_watched = friends[i]["watched"]
+        for i in range(len(friends_watched)):
+            if (friends_watched[i] not in user_watched) and (friends_watched[i] not in unique_unwatched):
+                unique_unwatched.append(friends_watched[i])
+
+    return unique_unwatched
+
+def get_available_recs(user_data):
+    recommended_movies = []
+    user_subscriptions = user_data["subscriptions"]
+    friends = user_data["friends"]
+    user_watched = user_data["watched"]
+    i = 0
+
+    for i in range(len(friends)):
+        friends_watched = user_data["friends"][i]["watched"]
+        for i in range(len(friends[i]["watched"])):
+            
+            if (friends_watched[i]["title"] not in user_watched) and (friends_watched[i]["host"] in user_subscriptions):
+                if friends_watched[i] not in recommended_movies:
+                    recommended_movies.append(friends_watched[i])
+
+    return recommended_movies
+
+def get_new_rec_by_genre(user_data):
+    genre_count = {}
+    recommended_movies = []
+    friends = user_data["friends"]
+    user_watched = user_data["watched"]
+    
+
+    if len(user_data["watched"]) == 0:
+        return recommended_movies
+
+    for item in user_data["watched"]:
+        if (item["genre"] in genre_count):
+            genre_count[item["genre"]] += 1
+        else:
+            genre_count[item["genre"]] = 1
+    
+    most_frequent_genre = max(genre_count, key=genre_count.get)
+
+    for i in range(len(friends)):
+        friends_watched = user_data["friends"][i]["watched"]
+        for i in range(len(friends[i]["watched"])):   
+            if (friends_watched[i]["title"] not in user_watched) and (friends_watched[i]["genre"] == most_frequent_genre):
```

Comment:
This could be approached similarly to the previous method. How could we start with a list of movies that the user hasn't seen, but at least one friend has?